src/commands/info/leaderboard.ts

```diff
@@ -2,38 +2,53 @@ import { SageUser } from '@lib/types/SageUser';
 import { Leaderboard } from '@root/src/lib/enums';
 import { Command } from '@lib/types/Command';
 import { createCanvas, CanvasRenderingContext2D, loadImage } from 'canvas';
-import { MessageEmbed, Message } from 'discord.js';
+import { MessageEmbed, ApplicationCommandOptionData, CommandInteraction } from 'discord.js';
 
 export default class extends Command {
 
 	description = 'Gives the top 10 users in the guild';
 	usage = '[page number]';
 	extendedHelp = 'Enter a page number to look further down the leaderboard';
 	runInDM = false;
-	aliases = ['rank', 'leader'];
 
-	async run(msg: Message, [page]: [number]): Promise<Message> {
-		msg.guild.members.fetch();
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'pagenumber',
+			description: 'leaderboard page to view',
+			type: 'NUMBER',
+			required: false
+		}
+	]
+
+	async tempRun(interaction: CommandInteraction): Promise<void> {
```

Comment:
It can take a few seconds to build the leaderboard image. You should probably differ this interaction since it make take longer than 3 seconds.